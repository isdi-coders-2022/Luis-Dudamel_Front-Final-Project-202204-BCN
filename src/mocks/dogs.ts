import { IDog } from "../interfaces/Dogs";

export const MockFavDogs: IDog[] = [
  {
    age: 7,
    name: "Rocko",
    picture: "rocko.jpg",
    title: "Ball Player",
    toy: "Ball",
    weight: "4",
    breed: "beagle",
    personality: "ball",
  },
  {
    age: 2,
    name: "Lamu",
    picture: "rocko.jpg",
    title: "Beach Lover",
    toy: "Ball",
    weight: "4",
    breed: "cocker",
    personality: "beach",
  },
  {
    age: 4,
    name: "Firulais",
    picture: "rocko.jpg",
    title: "Nature Explorer",
    toy: "Ball",
    weight: "4",
    breed: "border collie",
    personality: "nature",
  },
  {
    age: 4,
    name: "Breck",
    picture: "rocko.jpg",
    title: "Calm Walker",
    toy: "Ball",
    weight: "4",
    breed: "golden retriever",
    personality: "walker",
  },
  {
    age: 9,
    name: "Fenix",
    picture: "rocko.jpg",
    title: "Frisbee Player",
    toy: "Ball",
    weight: "4",
    breed: "collie",
    personality: "frisbee",
  },
  {
    age: 9,
    name: "Chuletas",
    picture: "rocko.jpg",
    title: "Calm Walker",
    toy: "Ball",
    weight: "4",
    breed: "beagle",
    personality: "walker",
  },
];
