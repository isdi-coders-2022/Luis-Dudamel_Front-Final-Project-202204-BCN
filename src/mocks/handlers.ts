import { rest } from "msw";

export const handlers = [
  rest.post(
    `${process.env.REACT_APP_API_URL_DEV}user/register`,
    (req, res, ctx) => {
      return res(ctx.status(201));
    }
  ),
];
