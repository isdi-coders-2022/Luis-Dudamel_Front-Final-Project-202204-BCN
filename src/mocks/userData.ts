export const mockToken: string =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiTHVpcyIsInVzZXJuYW1lIjoibHVpczEiLCJpZCI6IjYyOTY2MDM2Yjc0OTY5MjUxNDk2YmZmZCIsImlhdCI6MTY1NDE5ODQyNn0.H3MfL22YtoY3QFKSl11V7H_2DBhD6pB64O75Hllm2O8";
