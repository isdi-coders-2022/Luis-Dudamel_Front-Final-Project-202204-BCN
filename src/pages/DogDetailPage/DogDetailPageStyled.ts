import styled from "styled-components";

const StyledDogDetailPage = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 10px;
  margin-bottom: 10px;

  h1 {
    padding-top: 30px;
  }
`;
export default StyledDogDetailPage;
