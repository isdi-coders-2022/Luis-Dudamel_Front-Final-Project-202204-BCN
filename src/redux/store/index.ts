import { configureStore } from "@reduxjs/toolkit";
import uiSliceReducer from "../feature/usersSlice/uiSlice";
import userSliceReducer from "../feature/usersSlice/usersSlice";

const store = configureStore({
  reducer: {
    user: userSliceReducer,
    ui: uiSliceReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
