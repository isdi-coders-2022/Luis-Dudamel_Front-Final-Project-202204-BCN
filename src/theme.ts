import { createTheme } from "@mui/material/styles";
import { red } from "@mui/material/colors";

const theme = createTheme({
  typography: {
    fontFamily: ["Outfit"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "none",
          "&:hover": {
            backgroundColor: "#2a9d8f",
          },
        },
      },
    },
  },
  palette: {
    background: {
      default: "#ffc027",
    },
    primary: {
      main: "#264653",
    },
    secondary: {
      main: "#19857b",
    },
    error: {
      main: red.A400,
    },
  },
});

export default theme;
