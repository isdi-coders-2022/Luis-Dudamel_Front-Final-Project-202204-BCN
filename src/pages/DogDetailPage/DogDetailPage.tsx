import { useParams } from "react-router-dom";
import { DogDetail } from "../../components/DogDetail/DogDetail";
import { Header } from "../../components/Header/Header";
import { NavBar } from "../../components/NavBar/Navbar";
import { useAppSelector } from "../../redux/hooks";

export const DogDetailPage = (): JSX.Element => {
  const { id } = useParams();
  const currentDog = useAppSelector((state) =>
    state.dogs.find((dog) => dog.id === id)
  );

  return (
    <>
      <Header text={currentDog?.name} />
      <DogDetail dog={currentDog} />
      <NavBar />
    </>
  );
};
