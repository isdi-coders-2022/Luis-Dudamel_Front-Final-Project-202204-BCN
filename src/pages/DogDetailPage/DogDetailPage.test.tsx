import { ThemeProvider } from "@mui/material";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { mockFavDogs } from "../../mocks/dogs";

import store from "../../redux/store";
import theme from "../../theme";

import { DogDetailPage } from "./DogDetailPage";
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    id: "123",
  }),
}));

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

jest.mock("../../redux/hooks", () => ({
  ...jest.requireActual("../../redux/hooks"),
  useAppSelector: () => mockFavDogs,
}));

describe("Given a DogDetail component page", () => {
  describe("When it's invoked with a valid user id", () => {
    test("Then it should render a a button with the text 'Edit profile''", () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <DogDetailPage />
            </ThemeProvider>
          </BrowserRouter>
        </Provider>
      );

      const expectedButton = screen.getByRole("button", {
        name: "Edit profile",
      });

      expect(expectedButton).toBeInTheDocument();
    });
  });
});
