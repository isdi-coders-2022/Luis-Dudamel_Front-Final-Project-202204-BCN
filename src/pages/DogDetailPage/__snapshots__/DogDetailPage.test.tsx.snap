// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DogDetails component page When it's invoked Then it should always match this snapshot 1`] = `
Array [
  <div
    className="sc-dkzDqf ljjsOT"
  >
    <nav
      className="sc-gsnTZi cQdalN"
    >
      <div
        className="header-container MuiBox-root css-i9gxme"
      >
        <header
          className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic appbar css-18ditn3-MuiPaper-root-MuiAppBar-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              className="MuiTypography-root MuiTypography-h6 page-title css-poizw5-MuiTypography-root"
            >
              undefined's profile
            </div>
          </div>
        </header>
      </div>
    </nav>
    <div
      className="sc-bczRLJ kqivKm"
    >
      <div
        className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
      >
        <div
          className="dog-card-top"
        >
          <div
            className="image-container"
          >
            <img
              alt=" avatar"
              className="dog-card-top__avatar"
              crossOrigin=""
              src="http://localhost:4000/uploads/images/"
              width="100%"
            />
          </div>
          <img
            alt=" personality"
            className="dog-card-top__personality"
            src="../../images/icons/mobile/personalities/inactive/-inactive.png"
            width={20}
          />
        </div>
        <div
          className="dog-card-bottom"
        >
          <div>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
            />
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kqivKm dog-card-secondary-container"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="dog-card-secondary"
        >
          <div>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            >
              <span
                className="dog-card-bottom__bio-characteristic"
              >
                Age:
                 
              </span>
              <span
                className="dog-card-bottom__bio-property"
              >
                0
                 years
              </span>
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            >
              <span
                className="dog-card-bottom__bio-characteristic"
              >
                Breed:
                 
              </span>
              <span
                className="dog-card-bottom__bio-property"
              />
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            >
              <span
                className="dog-card-bottom__bio-characteristic"
              >
                Weight:
                 
              </span>
              <span
                className="dog-card-bottom__bio-property"
              >
                 kg
              </span>
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            >
              <span
                className="dog-card-bottom__bio-characteristic"
              >
                Favorite toy:
                 
              </span>
              <span
                className="dog-card-bottom__bio-property"
              />
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 dog-card-bottom__bio css-1cz1j3c-MuiTypography-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="bottom MuiBox-root css-1nl0624"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-gy4i07-MuiPaper-root"
    >
      <div
        className="MuiBottomNavigation-root css-1fj88io-MuiBottomNavigation-root"
      >
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-1vw7i77-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="Home navbar inactive icon"
            height="30px"
            onClick={[Function]}
            src="../../images/icons/mobile/navbar/inactive/home.png"
          />
          <span
            className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1ysd76y-MuiBottomNavigationAction-label"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-1vw7i77-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="Dates navbar inactive icon"
            height="30px"
            onClick={[Function]}
            src="../../images/icons/mobile/navbar/inactive/dates.png"
          />
          <span
            className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1ysd76y-MuiBottomNavigationAction-label"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-1vw7i77-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="Favorites navbar inactive icon"
            height="30px"
            onClick={[Function]}
            src="../../images/icons/mobile/navbar/inactive/friends.png"
          />
          <span
            className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1ysd76y-MuiBottomNavigationAction-label"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-1vw7i77-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="Profile navbar inactive icon"
            height="30px"
            onClick={[Function]}
            src="../../images/icons/mobile/navbar/inactive/profile.png"
          />
          <span
            className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1ysd76y-MuiBottomNavigationAction-label"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;
